mask.go

```diff
@@ -0,0 +1,53 @@
+// promconfig
+// Copyright 2020 Percona LLC
+//
+// Based on Prometheus systems and service monitoring server.
+// Copyright 2015 The Prometheus Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promconfig
+
+import "reflect"
+
+const maskedValue = "xxxxxxxx"
+
+// MaskSensitiveData loops over Config struct and masks sensitive data.
+func MaskSensitiveData(c interface{}) { //nolint:cyclop
```

Comment:
I wonder if we can use generics here to not use reference and return a value?
```suggestion
func MaskSensitiveData[T interface{}](c T) T { //nolint:cyclop
	nc, ok := deepcopy.Copy(c).(T)
	if !ok {
		return nil, errors.New("failed to copy config")
	}
        val := reflect.ValueOf(nc)
        ...
        return nc
```

Reply:
Yep, it will be nice

Reply:
@gen1us2k looks like we keep updating fields of an existing object. Is it possible to make it mutable to not update the existing object?

Reply:
Yes. But what problem do we want to solve? I think that probability of misusing Mask method is too low :)